app/models/company.rb

```diff
@@ -1,4 +1,6 @@
 # frozen_string_literal: true
 
 class Company < ApplicationRecord
+  has_many :shares, autosave: true, dependent: :destroy
```

Comment:
I'm not sure if you need `autosave: true` at this point. Maybe it can be added later when you really need it.

Reply:
I wasn't sure, it is not necessary for sure, will delete at this moment